### ISSUE TITLE
Array.isArray vs goog.isArray

### ISSUE THREAD

**Issue by fredj** [google/closure-compiler#1162]
The following code compiles without any error:
```js
// ==ClosureCompiler==
// @compilation_level ADVANCED_OPTIMIZATIONS
// @output_file_name default.js
// @use_closure_library true
// ==/ClosureCompiler==

/**
 * @param {Array.<string>|string} arg
 * @return {Array.<string>}
 */
function f(arg) {
  if (goog.isArray(arg)) {
    return arg;
  } else {
    return [arg];
  }
}
```

But if `goog.isArray` is replaced with `Array.isArray`:
```js
// ==ClosureCompiler==
// @compilation_level ADVANCED_OPTIMIZATIONS
// @output_file_name default.js
// @use_closure_library true
// ==/ClosureCompiler==

/**
 * @param {Array.<string>|string} arg
 * @return {Array.<string>}
 */
function f(arg) {
  if (Array.isArray(arg)) {
    return arg;
  } else {
    return [arg];
  }
}
```
The following warning is returned:
```
JSC_TYPE_MISMATCH: inconsistent return type
found   : (Array<string>|null|string)
required: (Array<string>|null) at line 7 character 11
    return arg;
```

Would it be possible to instruct the compiler about `Array.isArray`?

**Comment by MatrixFrog** [comment]
It looks like the compiler really understands goog.isArray, from here: https://github.com/google/closure-compiler/blob/0c09d82ab71f5518d6334692d279615e59c2b455/src/com/google/javascript/jscomp/type/ClosureReverseAbstractInterpreter.java#L185 for the old type inference. So we'd need to make sure there's a way it can understand non-`goog` things and then we could probably add in `Array.isArray` as well

**Comment by fredj** [comment]
[Number.isInteger()](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Number/isInteger) is an other good example

**Comment by Dominator008** [comment]
This is fixed in OTI and NTI: #1451 and #1452.
The following code:
```
/**
 * @param {!Array.<string>|string} arg
 * @return {Array.<string>}
 */
function f(arg) {
  if (Array.isArray(arg)) {
    return arg;
  } else {
    return [arg];
  }
}
```
compiles with NTI without warning. Note that `arg` should be declared as non-nullable.
